go/vendor/github.com/keybase/go-triplesec-insecure/triplesec.go

```diff
@@ -0,0 +1,352 @@
+// The design and name of TripleSec is (C) Keybase 2013
+// This Go implementation is (C) Filippo Valsorda 2014
+// Use of this source code is governed by the MIT License
+
+// Package triplesec implements the TripleSec v3 encryption and authentication scheme.
+//
+// For details on TripleSec, go to https://keybase.io/triplesec/
+package triplesec
+
+import (
+	"bytes"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/hmac"
+	"crypto/rand"
+	"crypto/sha512"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/keybase/go-triplesec/sha3"
+	"golang.org/x/crypto/salsa20"
+	"golang.org/x/crypto/scrypt"
+	"golang.org/x/crypto/twofish"
+)
+
+const SaltLen = 16
+
+type Cipher struct {
+	passphrase []byte
+	salt       []byte
+	derivedKey []byte
+}
+
+func scrub(b []byte) {
+	for i, _ := range b {
+		b[i] = 0
+	}
+}
+
+// A Cipher is an instance of TripleSec using a particular key and
+// a particular salt
+func NewCipher(passphrase []byte, salt []byte) (*Cipher, error) {
```

Comment:
The downside is that we're now forking the two triplesecs.  I guess that's not the end of the world.